framework/Objects/Meta_Fields.php

```diff
@@ -0,0 +1,266 @@
+	 * will be implicitly called when executing `unset($object->property)`.
+	 *
+	 * Note that if the property is not defined, this method will do nothing.
+	 * If the property is read-only, it will throw an exception.
+	 *
+	 * @param string $name The property name.
+	 *
+	 * @throws \Exception The property is read only.
+	 */
+	public function __unset( $name ) {
+		$setter = 'set_' . $name;
+		if ( method_exists( $this, $setter ) ) {
+			$this->$setter( null );
+		} elseif ( method_exists( $this, 'get_' . $name ) ) {
+			throw new \Exception( 'Setting read-only property: ' . get_class( $this ) . '::' . $name );
+		}
+	}
+
+	/**
+	 * Getter of postmeta from just custom fields
+	 *
+	 * @param string      $field_name Field name to get.
+	 * @param int         $post_id Post ID if different from get_the_ID.
+	 * @param bool|string $format_value Format value or not.
+	 *
+	 * @return mixed
+	 * @throws \Exception Unsupported custom fields plugin.
+	 */
+	protected function get_field_jcf( $field_name, $post_id, $format_value ) {
+		// Fix for getter from magic property field_*.
+		if ( strpos( $field_name, '_' ) !== 0 ) {
+			$field_name = "_{$field_name}";
+		}
+		return get_post_meta( $post_id, $field_name, $format_value );
+	}
+
+	/**
+	 * Getter of postmeta from advanced custom fields
+	 *
+	 * @param string      $field_name Field name to get.
+	 * @param int         $post_id Post ID if different from get_the_ID.
+	 * @param bool|string $format_value Format value or not.
+	 *
+	 * @return mixed
+	 * @throws \Exception Unsupported custom fields plugin.
+	 */
+	protected function get_field_acf( $field_name, $post_id, $format_value ) {
+		return get_field( $field_name, $post_id, $format_value );
+	}
+
+	/**
+	 * Main post meta fields getter function.
+	 *
+	 * @param string      $field_name Field name to get.
+	 * @param int         $post_id Post ID if different from get_the_ID.
+	 * @param bool|string $format_value Format value or not.
+	 *
+	 * @return mixed
+	 * @throws \Exception Unsupported custom fields plugin.
+	 */
+	public function get_field( $field_name, $post_id = null, $format_value = true ) {
+
+		// Use $this->post in case ID is empty.
+		if ( empty( $post_id ) && ! empty( $this->post ) ) {
+			$post_id = $this->post->ID;
+		}
+
+		// Check cache, if not exists - get field value.
+		if ( ! isset( $this->_fields[ $post_id ][ $field_name ] ) ) {
+			if ( 'just-custom-fields' === $this->custom_fields_plugin ) {
+				$value = $this->get_field_jcf( $field_name, $post_id, $format_value );
+			} elseif ( 'advanced-custom-fields' === $this->custom_fields_plugin ) {
+				$value = $this->get_field_acf( $field_name, $post_id, $format_value );
+			} else {
+				throw new \Exception( get_class( $this ) . "::get_field() : Unsupported custom fields plugin \"{$this->custom_fields_plugin}\"" );
+			}
+
+			$this->_fields[ $post_id ][ $field_name ] = $value;
+		}
+
+		return $this->_fields[ $post_id ][ $field_name ];
+	}
```

Comment:
get_field_jcf, get_field_acf - should be defined in child classes and should be abstract inside this class

Reply:
ready